### PR TITLE
Feature/new geometry v2

### PR DESCRIPTION
**Pull request by gustavogx**
Final (release) version of Geometry v2.

Cleaned up the gdml folder.

### CODE REVIEW

File: sbndcode/Geometry/gdml/README.txt [hunk 1]
```diff
@@ -1,18 +1,24 @@
...
 
-sbndvX(_nowires).gdml : Working geometry file for changes in progress. Do
-not use this gdml.
+$ preparseGDML sbnd_v02_00_base.gdml -w --setup DefaultWithShielding
 
+You can visualize the geometries with the geoVis_sbnd.C root macro:
+
+root 'geoVis_sbnd.C("sbnd_v02_00.gdml")'
+
+Pay attention to the fact that the _base files is not compatible with ROOT and cannot be visualided like this. It is, however, fully compatible with Geant4. If you venture to write a G4 visualizer, let me know.
```
Comment (ikatza) [inline sbndcode/Geometry/gdml/README.txt:24]: visualided --> visualized

File: sbndcode/Geometry/gdml/RefactoredGDMLSchema/gdml_define.xsd [hunk 2]
```diff
@@ -0,0 +1,118 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE xs:schema []>
+<xs:schema attributeFormDefault="unqualified" elementFormDefault="unqualified" version="1.0" xmlns:gdml="http://service-spi.web.cern.ch/service-spi/app/releases/GDML/schema" xmlns:xs="http://www.w3.org/2001/XMLSchema">
+  <xs:include schemaLocation="gdml_core.xsd"/>
+  <xs:include schemaLocation="gdml_extensions.xsd"/>
+  <!-- ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++ -->
+  
+  <xs:element name="define" type="defineType">
+    <xs:annotation>
+      <xs:documentation>Definition block of global named constants, quantitties, expressions,
```
Comment (ikatza) [inline sbndcode/Geometry/gdml/RefactoredGDMLSchema/gdml_define.xsd:10]: quantities

File: sbndcode/Geometry/gdml/geoVis_sbnd.C [hunk 3]
```diff
@@ -1,63 +1,214 @@
...
 		vol->SetTransparency(optsbnd[i].transparency);
 	}
+
+}
+
+
+struct wire{
+	TString Name;
+	float X,Y,Z;
+	wire(const TString &name, const float &x,const float &y,const float &z) : Name(name), X(x), Y(y), Z(z) {}
```
Comment (ikatza) [inline sbndcode/Geometry/gdml/geoVis_sbnd.C:96]: I think it's better to pass floats by value

**Comment by ikatza** [comment]
Had a quick read of some of the files. At the moment, I've only made tiny observations mostly related to spellchecking.

I've notice the PDS map is substantially different, I'll try to make a new image so someone with more knowledge can review it.

**Comment by gustavogx** [comment]
Hi Iker,

Thank you for your feedback.

About your comment on passing floats by value, yeah, I wasn't being careful
since this is a visualização macro, but you're right.

About the pds map, it should be the same cause Diego generated it a few
weeks ago. This is exactly the bug I was trying with to fix. Please take a
look at your logs and let me know if there are any warnings. There
shouldn't be any.

Best
Gustavo.

File: sbndcode/Geometry/gdml/sbnd_v02_00_base.gdml [hunk 4]
```diff
@@ -0,0 +1,6113 @@
...
+                        <fraction n="0.3707" ref="SiO2"/>
+                        <fraction n="0.0010" ref="P2O5"/>
+                        <fraction n="0.0248" ref="SO3"/>
+                        <fraction n="0.0034" ref="K2O"/>
+                        <fraction n="0.3865" ref="CaO"/>
+                        <fraction n="0.0044" ref="TiO2"/>
+                        <fraction n="0.0066" ref="MnO"/>
+                        <fraction n="0.0056" ref="Fe2O3"/>
+                        <fraction n="0.0001" ref="V2O5"/>
+                        <fraction n="0.0102" ref="sulphur"/>
```
Comment (ikatza) [inline sbndcode/Geometry/gdml/sbnd_v02_00_base.gdml:840]: It should be `sulfur` not `sulphur`, code follows US spelling not UK's.

**Comment by ikatza** [comment]
Hi @gustavogx the commit bf1d3ce24 removed some essential code.

At least I found this to be necessary: 

```
		<material name="P2O5" formula="P2O5">
 			<D value="1.562" unit="g/cm3"/>
 			<composite n="2" ref="phosphorous"/>
 			<composite n="5" ref="oxygen"/>
 		</material>
 		<material name="SO3" formula="SO3">
 			<D value="1.92" unit="g/cm3"/>
 			<composite n="1" ref="sulfur"/>
 			<composite n="3" ref="oxygen"/>
 		</material>
 		<material name="K2O" formula="K2O">
 			<D value="2.35" unit="g/cm3"/>
 			<composite n="1" ref="hydrogen"/>
 			<composite n="1" ref="oxygen"/>
 			<composite n="2" ref="potassium"/>
 		</material>
 		<material name="MnO" formula="MnO">
 			<D value="5.37" unit="g/cm3"/>
 			<composite n="1" ref="manganese"/>
 			<composite n="1" ref="oxygen"/>
 		</material>
 		<material name="V2O5" formula="V2O5">
 			<D value="3.36" unit="g/cm3"/>
 			<composite n="1" ref="vanadium"/>
 			<composite n="5" ref="oxygen"/>
 		</material>
```

**Comment by ikatza** [comment]
I believe the PDS Map is alright, attaching plots: 
sbnd_pds_map_tpc0
sbnd_pds_map_tpc1

**Comment by ikatza** [comment]
I've produced a few events using this last commit, both with the regular fcl and with refactored ones, and all seems good.

Note that CI will most likely fail.

The new PDS map is a breaking change. Since the number of optical channels changed, along with the optical detector associated to the channel, I suspect there will be an exception when trying to obtain data for certain optical channels.

Similarly because the change of geometry, art should throw an exception when using files created with the old geometry as input.

**Comment by ikatza** [comment]
I understand there were some merge conflicts, so @marcodeltutto should have a final check.

**Comment by ikatza** [comment]
Forgot to mention, this solves #75

**Comment by ikatza** [comment]
approve

**Comment by marcodeltutto** [comment]
Just checked and there are no more merge conflicts. All looks good to me!

@ikatza is correct that this is a breaking change and art will throw an exceptional when running over larsoft files made with previous versions. @pgreen135 I think it would be nice to mention this in the announcement email with the sbndcode version that will contain this PR

**Comment by pgreen135** [comment]
thanks all, I will include it in the next release and will mention the breaking change in the announcement

**Comment by pgreen135** [comment]
trigger build SBNSoftware/sbndcode#119

**Comment by pgreen135** [comment]
@gustavogx @marcodeltutto my understanding was that this PR is to replace the v01_05 geometry with v2 by default, however the CI tests (which just run the standard g4 fhicl, etc.) appear to be using the older geometry. Is this instead only for the refactored larg4? 

also disregard the c7 build error, this is due to an issue upstream in sbncode

**Comment by marcodeltutto** [comment]
Hi @pgreen135, the geometry v2 is intended for both the legacy and the refactored larg4. I see that the CI tests are using the v01_05 geometry, which I didn't expect. If I checkout this branch I get:
```
fhicl-dump sbnd_ci_single_gen_quick_test_sbndcode.fcl | grep "GDML"
      GDML: "sbnd_v02_00.gdml"
      GDML: "sbnd_v02_00.gdml"
```

And if I run that fcl I see that is using the v2 geometry.

Looking at the CI [checkout](https://dbweb8.fnal.gov:8443/LarCI/app/ns:SBND/build_detail/phase_details?build_id=sbnd_ci/6592&platform=Linux%203.10.0-1160.31.1.el7.x86_64&phase=checkout_github&buildtype=slf7%20e20:prof) and build stages, it looks to me that the CI is not building this branch, but is building `sbndcode` at commit bd606d17. Or am I misinterpreting this?

**Comment by pgreen135** [comment]
Hi @marcodeltutto yes I think you are right, this doesn't appear to have built as intended -- I will try again without Iker's branch also included

**Comment by pgreen135** [comment]
trigger build

**Comment by pgreen135** [comment]
@marcodeltutto looks like it's now working as expect, sorry for the false alert! Interesting now fails the unit tests as they have always been running with completely the wrong now removed geometry D: I'll see if I can fix that...

**Comment by pgreen135** [comment]
digging into this further -- the first two failed unit tests are due to being hard-coded to read geometry version v01_00, easily fixed and now pass. The other tests however are reading the correct geometry but failing with this error: 

%MSG-i GeometryTest:  GeometryTest:geotest@BeginJob 02-Jul-2021 06:34:09 CDT  ModuleBeginJob GeometryTestAlg.cxx:214
test FindAllVolumes method ...
%MSG
Found 0 world volumes 'volWorld':
%MSG-e GeometryTest:  GeometryTest:geotest@BeginJob 02-Jul-2021 06:34:09 CDT  ModuleBeginJob
Found 0 world volumes 'volWorld! [expecting: one!!]
%MSG

@gustavogx @marcodeltutto could you look into this?

The full logs are available at: https://dbweb8.fnal.gov:8443/LarCI/app/ns:SBND/build_detail/phase_details?build_id=sbnd_ci/6596&platform=Linux%203.10.0-1160.31.1.el7.x86_64&phase=unit_test&buildtype=slf7%20e20:prof [relevant ones are tests 3 and 4)

**Comment by marcodeltutto** [comment]
Thanks @pgreen135! They are failing because there are two setups in the gdml file:
```
	<setup name="Default" version="1.0">
		<world ref="volWorld" />
	</setup>
 	<setup name="DefaultWithShielding" version="1.0">
		<world ref="volWorldWithShielding" />
	</setup>
```
Removing the second one (which is not needed for the standard gdmls) will fix this problem. This needs to be changed in the `preparseGDML`, which generates the gdml files. I can look into this.

**Comment by marcodeltutto** [comment]
@pgreen135 The last commits should fix the unit tests

**Comment by pgreen135** [comment]
Thanks @marcodeltutto, and thanks for fixing the hard-coding!

**Comment by pgreen135** [comment]
trigger build

**Comment by pgreen135** [comment]
trigger build

**Pull request merged**